### ISSUE TITLE
Documentation: Code Entry Point and Data Preprocessing Details for `B13_rn18_moco_0099_ckpt.pth`

### ISSUE THREAD

**Issue by douglasmacdonald** [zhu-xlab/SSL4EO-S12#28]
Thank you for providing the pre-trained weights B13_rn18_moco_0099_ckpt.pth. 

Could you please specify the exact code entry point used to generate these weights? The codebase includes multiple data-loaders, and I want to ensure that I replicate the same normalization, rescaling, and other preprocessing steps applied during the training of these weights.

Based on your paper, the weights seem to have been generated after converting images to uint8 format. Can you confirm whether the images were rescaled to the same size during the training process for these weights? 

I understand that upscaling was only applied for downstream tasks. If you had the time, would you have done this for these weights, too?

**Comment by wangyi111** [comment]
Hi! Thx for your interest! The main code for generating the moco weights (rn18/50) is [pretrain_moco_v2_s2c.py](https://github.com/zhu-xlab/SSL4EO-S12/blob/main/src/benchmark/pretrain_ssl/pretrain_moco_v2_s2c.py). The script we used was [this one](https://github.com/zhu-xlab/SSL4EO-S12/blob/main/src/benchmark/pretrain_ssl/scripts/pretrain/srun_train_moco_rn50_s2c.sh). This itself is rn50, for rn18 simply change `--arch resnet50` to "--arch resnet18".

For sentinel 2 weights in this paper, we divide the int16 values by 10000 to (0,1), convert to uint8, and that's it for preprocessing. There are some data augmentations for the pretraining, you can see it in [`pretrain_moco_v2_s2c.py`](https://github.com/zhu-xlab/SSL4EO-S12/blob/2156913c5d8e5a2c572a5b000f0d5eaed6fc3192/src/benchmark/pretrain_ssl/pretrain_moco_v2_s2c.py#L327). Regarding image sizes specifically, we simply perform random resized cropping to 224x224 (raw is 264x264).

Yep upscaling was only applied for downstream tasks. According to our limited exploration (Appendix-3-B), upscaling is more effective and efficient in downstream than in pretraining (it costs much more compute in pretraining, and you will need to upscale the downstream again to get the benefit). We can't tell now for sure if it's worth that, but once we confirm with more thorough analysis we will update! If you have any good references also welcome to share with us (e.g. DINOv2 has an progressive/short-term upscaling to 416 which is as good as full fine-resolution pretraining, this could be a good idea for our study).

**Comment by wangyi111** [comment]
Just add to the normalization, since dividing 10000 will result in many images with many bands (e.g. RGB) falling in (0,0.5) and eventually (0,128) in uint8, the performance might be limited. Some further works using our dataset (e.g. [CROMA](https://github.com/antofuller/CROMA)) perform mean/std based normalization and it might work better in many downstream tasks.

**Comment by douglasmacdonald** [comment]
Apologies for the delayed response, but thank you! I only occasionally dabble with this in my spare time.

**Comment by wangyi111** [comment]
Hope you have fun!
